src/rules/noUnusedVariableRule.ts

```diff
@@ -225,7 +225,14 @@ function addImportSpecifierFailures(ctx: Lint.WalkContext<void>, failures: Map<t
         }
 
         function removeAll(errorNode: ts.Node, failure: string): void {
-            const fix = Lint.Replacement.deleteFromTo(importNode.getStart(), importNode.getEnd());
+            const start = importNode.getStart();
+            let end = importNode.getEnd();
+            if (ctx.sourceFile.getLineAndCharacterOfPosition(start).character === 0 &&
+                ctx.sourceFile.getLineEndOfPosition(end) === end) {
+                end++;
```

Comment:
that will only work if you're using `\n`. if there's `\r\n`, it will only remove `\r`

alternative to `end++`:
```ts
const nextLine = ctx.sourceFile.getLineAndCharacterOfPosition(end).line + 1;
const lineStarts = ctx.sourceFile.getLineStarts();
if (lineStarts.length !== nextLine) {
    end = lineStarts[nextLine];
}
```